src/main/java/eu/greev/dcbot/ticketsystem/service/Transcript.java

```diff
@@ -1,135 +1,114 @@
 package eu.greev.dcbot.ticketsystem.service;
 
-import eu.greev.dcbot.ticketsystem.entities.Ticket;
+import eu.greev.dcbot.ticketsystem.entities.Edit;
+import eu.greev.dcbot.ticketsystem.entities.Message;
+import eu.greev.dcbot.ticketsystem.entities.TranscriptEntity;
 import lombok.Getter;
+import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
+import org.apache.logging.log4j.util.Strings;
 
-import java.io.*;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Date;
 import java.util.List;
 
 @Slf4j
+@Getter
+@RequiredArgsConstructor
 public class Transcript {
-    private final int id;
-    @Getter private final File transcript;
+    private final List<TranscriptEntity> recentChanges = new ArrayList<>();
+    private final List<Message> messages;
 
-    public Transcript(Ticket ticket) {
-        id = ticket.getId();
+    public void addMessage(net.dv8tion.jda.api.entities.Message message, int ticketId) {
+        Message msg = new Message(message.getIdLong(), message.getContentDisplay(), message.getAuthor().getName(), message.getTimeCreated().toEpochSecond(), ticketId);
+        messages.add(msg);
+        recentChanges.add(msg);
+    }
+
+    public void addLogMessage(String log, long timestamp, int ticketId) {
+        Message message = new Message(0, log, "", timestamp, ticketId);
+        messages.add(message);
+        recentChanges.add(message);
+    }
+
+    public void editMessage(long messageId, String content, long timeEdited) {
+        Edit edit = new Edit(content, timeEdited, messageId);
+
+        messages.stream()
+                .filter(m -> m.getId() == messageId)
+                .findFirst().ifPresent(m -> m.getEdits().add(edit));
+        recentChanges.add(edit);
+    }
+
+    public void deleteMessage(long messageId) {
+        messages.stream()
+                .filter(m -> m.getId() == messageId)
+                .findFirst().ifPresent(m -> m.setDeleted(true));
+        recentChanges.stream()
+                .filter(Message.class::isInstance)
+                .filter(m -> (((Message) m).getId()) == messageId)
+                .findFirst().ifPresentOrElse(m -> ((Message) m).setDeleted(true), () -> {
+                    Message message = new Message(messageId, "", "", 0, 0);
```

Comment:
Why add an empty message to the recent changes? Can we just leave it out, or would that create a bug?

Reply:
Also, since the timestamp is 0 shouldn't it be at the very start of all recent changes

Reply:
This object only represents the change of the message (in case its edited or deleted), so I dont need the content / author / timestamp / ticketId
I just set this message to `isDeleted();` to send this change to the database